src/Messenger/IndexationRequestHandler.php

```diff
@@ -26,27 +28,58 @@ abstract class IndexationRequestHandler implements MessageHandlerInterface
 
     private $client;
 
-    public function __construct(Client $client)
+    private $bus;
+
+    public function __construct(Client $client, MessageBusInterface $bus)
     {
         $this->client = $client;
+        $this->bus = $bus;
 
         // Disable the logs for memory concerns
         $this->client->setLogger(new NullLogger());
     }
 
     public function __invoke(IndexationRequestInterface $message)
     {
-        $indexer = $this->client->getIndexer();
-
+        $messages = [];
         if ($message instanceof MultipleIndexationRequest) {
-            foreach ($message->getOperations() as $operation) {
-                $this->schedule($indexer, $operation);
-            }
+            $messages = $message->getOperations();
         } elseif ($message instanceof IndexationRequest) {
-            $this->schedule($indexer, $message);
+            $messages = [$message];
         }
 
-        $indexer->flush();
+        $indexer = $this->client->getIndexer();
+        $originalBulkMaxSize = $indexer->getBulkMaxSize();
+        $initialQueueSize = $indexer->getQueueSize();
+
+        try {
+            $indexer->setBulkMaxSize($initialQueueSize + count($messages));
```

Comment:
I'm not a big fan of this. 

Messenger Handler can be used in a variety of setup including sync, and this will change the Indexer behavior application-wide. Also it drops the threshold, the Max Bulk Size is necessary to avoid sending huge payload over the network (what if I have 6000 messages in my MultipleIndexationRequest?).

It can be done differently;
let the Indexer "flush" whenever it wants, and you will still get an Exception.

When that occurs, we know where we are in the foreach loop:
- what messages have been sent (and could be in error)
- what messages are not sent and can be processed next

WDYT?

Reply:
I start working in this in b673c446498c419a50e46a72037a73615319e1c0

Anyway it doesn't cover yet the case a non-first bulk triggers an exception

Reply:
finished via e8b0918d122c3533e0ba9d064b813d95a22eb103